src/include/txn.i

```diff
@@ -218,37 +222,20 @@ __wt_txn_resolve_prepared_op(
 	    (WT_CURSOR_BTREE *)cursor, &upd));
 	WT_ERR(ret);
 
-#ifdef HAVE_DIAGNOSTIC
 	/*
-	 * Do what we can to ensure that finding prepared updates from a key
-	 * is working as expected. In the case where a transaction has updated
-	 * the same key multiple times, it's possible to resolve all updates
-	 * for the key when processing the first op structure, and then have
-	 * eviction free those updates before subsequent ops are processed,
-	 * which means a search could reasonably not find an update in that
-	 * case.
-	 * We track the update count only for commit, but not for rollback, as
-	 * our tracking is based on transaction id, and in case of rollback, we
-	 * set it to aborted.
+	 * If we haven't found anything then there's an error.
 	 */
-	if (upd == NULL && commit) {
-		/*
-		 * FIXME:
-		 * WT_ASSERT(session, txn->multi_update_count > 0);
-		 */
-		--txn->multi_update_count;
+	if (upd == NULL) {
+		WT_ASSERT(session, upd != NULL);
+		WT_ERR(WT_NOTFOUND);
 	}
-#endif
-
-	WT_STAT_CONN_INCR(session, txn_prepared_updates_resolved);
-
-	for (; upd != NULL; upd = upd->next) {
-		 if (upd->txnid != txn->id)
-			continue;
 
+	for (; upd != NULL && upd->txnid == txn->id; upd = upd->next) {
```

Comment:
Looks like we are stopping at the update that doesn't correspond to this transaction id `txn->id`. Earlier we used to go past it and search for updates that might match `txn->id`. Are we saying that once we have updates from another transaction in the update chain, we can not have updates from the current transaction `txn->id`. I think this will be correct, wouldn't hurt double checking with another reviewer.